test/e2e/network_segmentation.go

```diff
@@ -364,6 +364,178 @@ var _ = Describe("Network Segmentation", func() {
+						ObjectMeta: metav1.ObjectMeta{
+							Name: namespace,
+						},
+					}, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					defer func() {
+						Expect(cs.CoreV1().Namespaces().Delete(
+							context.Background(),
+							namespace,
+							metav1.DeleteOptions{},
+						)).To(Succeed())
+					}()
+				}
+				networkNamespaceMap := map[string]string{namespaceRed: red, namespaceBlue: blue}
+				for namespace, network := range networkNamespaceMap {
+					By("creating the attachment configuration for network " + network + " in namespace " + namespace)
+					netConfig := newNetworkAttachmentConfig(nad)
+					netConfig.namespace = namespace
+					netConfig.name = network
+
+					_, err = nadClient.NetworkAttachmentDefinitions(namespace).Create(
+						context.Background(),
+						generateNAD(netConfig),
+						metav1.CreateOptions{},
+					)
+					Expect(err).NotTo(HaveOccurred())
+				}
+				pods := []*v1.Pod{}
+				redIPs := []string{}
+				blueIPs := []string{}
+				for namespace, network := range networkNamespaceMap {
+					for i := range numberOfPods {
+						podConfig := *podConfig(
+							fmt.Sprintf("%s-pod-%d", network, i),
+							withCommand(func() []string {
+								return httpServerContainerCmd(port)
+							}),
+						)
+						podConfig.namespace = namespace
+						//ensure testing accross nodes
+						if i%2 == 0 {
+							podConfig.nodeSelector = map[string]string{nodeHostnameKey: workerOneNodeName}
+
+						} else {
+
+							podConfig.nodeSelector = map[string]string{nodeHostnameKey: workerTwoNodeName}
+						}
+						By("creating pod " + podConfig.name + " in " + podConfig.namespace)
+						pod := runUDNPod(cs, podConfig.namespace, podConfig, nil)
+						pods = append(pods, pod)
+						podIP, err := podIPsForUserDefinedPrimaryNetwork(
+							cs,
+							pod.Namespace,
+							pod.Name,
+							namespacedName(namespace, network),
+							0,
+						)
+						Expect(err).NotTo(HaveOccurred())
+						if network == red {
+							redIPs = append(redIPs, podIP)
+						} else {
+							blueIPs = append(blueIPs, podIP)
+						}
+					}
+				}
+
+				By("ensuring pods only communicate with pods in their network")
+				for _, pod := range pods {
+					isRedPod := strings.Contains(pod.Name, red)
+					ips := redIPs
+					if !isRedPod {
+						ips = blueIPs
+					}
+					for _, ip := range ips {
+						result, err := e2ekubectl.RunKubectl(
+							pod.Namespace,
+							"exec",
+							pod.Name,
+							"--",
+							"curl",
+							"--connect-timeout",
+							"2",
+							net.JoinHostPort(ip, fmt.Sprintf("%d", port)+"/hostname"),
+						)
+						Expect(err).NotTo(HaveOccurred())
+						if isRedPod {
+							Expect(strings.Contains(result, red)).To(BeTrue())
+						} else {
+							Expect(strings.Contains(result, blue)).To(BeTrue())
+						}
+					}
+				}
+
+				By("Deleting pods in network blue except " + fmt.Sprintf("%s-pod-%d", blue, numberOfPods-1))
```

Comment:
maybe you should ensure there's a pod left in the namespace after deleting all but one ? 

Not really sure it's worth it. Up to you.

Reply:
when you get the ip for the UDN pod if the pod does not exist it will error. Is that enough?

Reply:
hmmmm maybe. I would prefer to have an explicit check, but if the error message is clear enough, it would be OK.